### DISCUSSION TITLE
How to use anchors and mathematical grammar

### DISCUSSION THREAD

**Discussion by LookForwordTomorrow** [remarkjs/remark#610] [Q&A] [answered]
After adding an anchor point to the markdown file, I found that the a tag has a point of # xxxxx, but the target tag does not have an ID. I noticed that the official website has a custom mathematical grammar, but there is no way to use it directly. I don't know what grammar he will use.

**Comment by ChristianMurphy** [comment, marked as answer]
Welcome @LookForwordTomorrow :wave: 

> After adding an anchor point to the markdown file, I found that the a tag has a point of # xxxxx, but the target tag does not have an ID

could you provide an example of what input markdown you tried, and a sample of the output (HTML, Markdown, or AST) you want to get?

> noticed that the official website has a custom mathematical grammar, but there is no way to use it directly. I don't know what grammar he will use.

Math syntax can be enabled with `remark-math` https://github.com/remarkjs/remark-math
Would that work for you?

**Reply by LookForwordTomorrow** [reply]
I integrated remark math according to the official example, but it didn't succeed. The syntax was still displayed in string text format instead of mathematical grammar text

**Reply by ChristianMurphy** [reply]
@LookForwordTomorrow It sounds like you have [tried something](https://github.com/remarkjs/remark/discussions/610#discussioncomment-297970) but it isn't working as expected?
Could you share a runnable example of the issue you are seeing? https://codesandbox.io/s/react-markdown-debug-9n4eg is a good starting point.
And explain some more what you expect the output to be?

**Reply by ChristianMurphy** [reply]
@LookForwordTomorrow here is a codesandbox with the example from the readme https://codesandbox.io/s/react-markdown-debug-forked-hud82
It appears to be working.
With missing styles you may not have imported the CSS

```js
import 'katex/dist/katex.min.css' // `react-katex` does not import the CSS for you
```

**Reply by LookForwordTomorrow** [reply]
![image](https://user-images.githubusercontent.com/40135607/105277874-4a75c980-5bdf-11eb-9181-83be10f0d4ce.png)

**Reply by ChristianMurphy** [reply]
Right, react-markdown is [CommonMark](https://commonmark.org) by default, if you want to add extra features, like math, you need to use plugins.
Again see the fully functional example I shared in code sandbox https://codesandbox.io/s/react-markdown-debug-forked-hud82

**Reply by LookForwordTomorrow** [reply]
![image](https://user-images.githubusercontent.com/40135607/105283483-86169080-5beb-11eb-83eb-59b246e22920.png)
The introduction of CSS error, like a path problem, invalid font path

**Reply by LookForwordTomorrow** [reply]
Can show, but a lot of grammar can not be converted, because I format wrong? This kind of start with is invalid
![image](https://user-images.githubusercontent.com/40135607/105329336-01883a00-5c0c-11eb-851c-aaf7898b6fdb.png)

**Reply by wooorm** [reply]
Math in markdown uses dollar signs around math in markdown: https://github.com/remarkjs/remark-math

**Reply by LookForwordTomorrow** [reply]
Thanks for your reply, the grammar is ready to use.

But I still have a question, if you want to realize the anchor jump, how should it be realized?

There is no corresponding ID attribute on H1 ~ H6 tags. How should I configure it。

**Reply by wooorm** [reply]
For that, I refer to @ChristianMurphy’s previous comment about that:

> > After adding an anchor point to the markdown file, I found that the a tag has a point of # xxxxx, but the target tag does not have an ID
>
> could you provide an example of what input markdown you tried, and a sample of the output (HTML, Markdown, or AST) you want to get?

Because it’s still unclear to us what you mean

**Comment by LookForwordTomorrow** [comment]
![image](https://user-images.githubusercontent.com/40135607/105275473-6dea4580-5bda-11eb-92a6-42a0c6572e62.png)
![image](https://user-images.githubusercontent.com/40135607/105275562-9c682080-5bda-11eb-9678-567e75e83b19.png)

**Reply by ChristianMurphy** [reply]
Okay...?
What's your question?
Doesn't really answer either of follow up clarifications from https://github.com/remarkjs/remark/discussions/610#discussioncomment-297953

**Reply by LookForwordTomorrow** [reply]
markdown file
![image](https://user-images.githubusercontent.com/40135607/105468211-b5073200-5cd1-11eb-88c5-479499d6a633.png)
design sketch
![image](https://user-images.githubusercontent.com/40135607/105468398-e8e25780-5cd1-11eb-9161-ec3ed75120af.png)
Click A1 to scroll to the A1 title section, but there is no jump

**Reply by wooorm** [reply]
You can define your own [`renderers`](https://github.com/remarkjs/react-markdown#api) to take `heading`s and add IDs to them.

An alternative is that https://github.com/remarkjs/remark-slug should enable that. But if that doesn‘t work, it’ll work soon: https://github.com/remarkjs/react-markdown/issues/427#issuecomment-752414256.
